wpe/src/main/java/com/wpe/wpeview/WPECookieManager.java

```diff
@@ -0,0 +1,110 @@
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+package com.wpe.wpeview;
+
+import android.annotation.SuppressLint;
+import android.os.Handler;
+
+import androidx.annotation.Keep;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public final class WPECookieManager {
+
+    private enum NativeCookieAcceptPolicy {
+        AcceptAlways(0),
+        AcceptNever(1),
+        AcceptNoThirdParty(2);
+
+        private final int value;
+
+        NativeCookieAcceptPolicy(int value) { this.value = value; }
+
+        public int getValue() { return value; }
+    }
+
+    private static final WPECookieManager instance = new WPECookieManager();
+
+    // Keep these local because fetching them is async in wpe
+    private boolean acceptCookie = true;
+    private boolean acceptThirdpartyCookie = false;
+
+    private WPECookieManager() {}
+
+    public static @NonNull WPECookieManager getInstance() { return instance; }
+
+    public boolean getAcceptCookie() { return acceptCookie; }
+
+    public void setAcceptCookie(boolean accept) {
+        if (acceptCookie == accept)
+            return;
+
+        acceptCookie = accept;
+
+        NativeCookieAcceptPolicy policy = NativeCookieAcceptPolicy.AcceptNever;
+        if (accept) {
+            if (!acceptThirdpartyCookie) {
+                policy = NativeCookieAcceptPolicy.AcceptNoThirdParty;
+            } else {
+                policy = NativeCookieAcceptPolicy.AcceptAlways;
+            }
+        }
+
+        nativeSetCookieAcceptPolicy(policy.getValue());
+    }
+
+    public boolean getAcceptThirdPartyCookies() { return acceptThirdpartyCookie; }
+
+    public void setAcceptThirdPartyCookies(boolean accept) {
+        if (acceptThirdpartyCookie == accept)
+            return;
+
+        acceptThirdpartyCookie = accept;
+
+        // Update wpe only we accept cookies in general
+        if (acceptCookie) {
+            NativeCookieAcceptPolicy policy = acceptThirdpartyCookie ? NativeCookieAcceptPolicy.AcceptAlways
+                                                                     : NativeCookieAcceptPolicy.AcceptNoThirdParty;
+            nativeSetCookieAcceptPolicy(policy.getValue());
+        }
+    }
+
+    public void removeAllCookies(@Nullable Callback callback) {
+        if (callback != null) {
+            nativeRemoveAllCookies(callback);
+        }
```

Comment:
After re-reading this I was wondering if it would not be more interesting to still call `nativeRemoveAllCookies()` with a default callback doing nothing if `callback` is null (or check for a null callback on the native side either).

Because you may have a caller which wants to remove all cookies but which doesn't care about when the request has finished execution.

Reply:
Makes sense